src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -97,15 +97,14 @@ private void LoadManifestsFromProvider(IWorkloadManifestProvider manifestProvide
                 using (var manifestStream = openManifestStream())
                 {
                     var manifest = WorkloadManifestReader.ReadWorkloadManifest(manifestId, manifestStream);
-                    if (_manifests.ContainsKey(manifestId))
+                    if(!string.Equals (manifestId, manifest.Id, StringComparison.OrdinalIgnoreCase))
                     {
-                        throw new Exception($"Duplicate workload manifest {manifestId}");
+                        throw new WorkloadManifestCompositionException($"Manifest '{manifestId}' from provider {manifestProvider} does not match payload id '{manifest.Id}'");
```

Comment:
They probably should, yes. I didn't worry because it's possible they'll never be seen, and they aren't very useful to end-users. They're basically internal errors and indicate either the manifests themselves are broken or the combination of manifests is broken.

That said, a lot of the workload manifest errors are localized, so these probably should be too for consistency.

Perhaps the CLI should catch `WorkloadManifestCompositionException` and `WorkloadManifestFormatException` and report an error that says the installation (or nuget sources) is broken and suggests repairing or reinstalling.